### PR TITLE
Create code object tab as module global with indices

### PR DESCRIPTION
**Pull request by da-woods**
This currently looks to only save a tiny amount of space so may not actually be worth it. Therefore I'm marking it as draft.

The one useful thing it does do is to restore some error origin information that was lost from the code object generation. However we could cherry-pick it anyway.

It also might make it easier to pack the structure better and so save some actual space.

### CODE REVIEW

File: Cython/Compiler/ExprNodes.py [hunk 1]
```diff
@@ -10156,6 +10156,10 @@ class CodeObjectNode(ExprNode):
     is_temp = False
     result_code = None
 
+    tuple_index_re = re.compile(
+        rf"{Naming.pyrex_prefix}tuple\[(\d+)\]"
```
Comment (da-woods) [inline Cython/Compiler/ExprNodes.py:10160]: There's definitely better ways of doing that. I just don't really want to spend too much time on it until we've decided if this is worthwhile or should be abandoned as a marginal gain.

**Comment by scoder** [comment]
We shouldn't be using prebuilt tuples for the argument names. They are really only used once and should not take up any space in the module state.

**Comment by da-woods** [comment]
> We shouldn't be using prebuilt tuples for the argument names. They are really only used once and should not take up any space in the module state.

Not convinced this is necessarily always right.

`(self, )` is likely pretty common. `(self, env)`, `(self, code)`, and `(self, node)` will occur loads of times in the Cython code-base.

However - given many are only used once - I actually wonder if we should be reverting the whole "code object table" change? It made a pretty minor size difference anyway, and it makes it harder to be flexible about things like "mixture of prebuilt and non-prebuild tuples". (We should keep the cached code objects stored in an array on the module state rather than many individual constants though)

**Comment by scoder** [comment]
I didn't suggest to stop deduplicating these tuples, just that they should not end up in the module state. We don't need a permanent reference to them after the module init phase. And for the few that we still happen to need later, it's probably fine to use a copy.

But yes, of it doesn't pay off, there's no use in throwing more complexity at them.

**Pull request closed without merging**